models/yang/sonic/sonic-acl.yang

```diff
@@ -0,0 +1,233 @@
+				}
+			}
+		}
+
+		container ACL_RULE {
+
+			list ACL_RULE_LIST { 
+				key "aclname rulename";
+
+				leaf aclname {
+					type leafref {
+						path "../../../ACL_TABLE/ACL_TABLE_LIST/aclname";
+					}
+				} 
+
+				leaf rulename {
+					type string {
+						pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{1,63})' {
+							error-app-tag rulename-invalid;
+						}
+					}
+				}
+
+				leaf PRIORITY {
+					type uint16 {
+						range "1..65535"{
+							error-app-tag priority-invalid-range;
+							error-message "Invalid ACL rule priority.";
+						}
+					}
+				}
+
+				leaf RULE_DESCRIPTION {
+					type string {
+						length 1..255 {
+							error-app-tag ruledesc-invalid-length;
+						}
+					}
+				}
+
+				leaf PACKET_ACTION {
+					mandatory true;
+					type enumeration {
+						enum FORWARD;
+						enum DROP;
+						enum REDIRECT;
+					}
+				}
+
+				leaf MIRROR_ACTION {
+					type string {
+						pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{1,63})' {
+							error-app-tag mirror-action-invalid;
+						}
+					}
+				}
+
+				leaf IP_TYPE {
+					mandatory true;
+					type enumeration {
+						enum ANY;
+						enum IP;
+						enum IPV4;
+						enum IPV4ANY;
+						enum NON_IPV4;
+						enum IPV6ANY;
+						enum NON_IPV6;
+					}
+				}
+
+				leaf IP_PROTOCOL {
+					type uint8 {
+						range "1|2|6|17|46|47|51|103|115";
+					}
+				}
+
+				leaf ETHER_TYPE {
+					type string {
+						pattern "(0x88CC)|(0x8100)|(0x8915)|(0x0806)|(0x0800)|(0x86DD)|(0x8847)" {
+							error-message "Invalid ACL Rule Ether Type";
+							error-app-tag ether-type-invalid;
+						}
+					}
+				}
+
+				choice ip_src_dst {
+					case ipv4_src_dst {
+						when "boolean(IP_TYPE[.='ANY' or .='IP' or .='IPV4' or .='IPV4ANY'])";
+						leaf SRC_IP {
+							mandatory true;
+							type inet:ipv4-prefix;
+						}
+						leaf DST_IP {
+							mandatory true;
+							type inet:ipv4-prefix;
+						}
+					}
+					case ipv6_src_dst {
+						when "boolean(IP_TYPE[.='ANY' or .='IP' or .='IPV6' or .='IPV6ANY'])";
+						leaf SRC_IPV6 {
+							mandatory true;
+							type inet:ipv6-prefix;    
+						}
+						leaf DST_IPV6 {
+							mandatory true;
+							type inet:ipv6-prefix;    
+						}
+					}
+				}
+
+				choice src_port {
+					case l4_src_port {
+						leaf L4_SRC_PORT {
+							type uint16;
+						}
+					}
+					case l4_src_port_range {
+						leaf L4_SRC_PORT_RANGE {
+							type string {
+								pattern "[0-9]{1,5}(-)[0-9]{1,5}" {
+									error-app-tag src-port-range-invalid;
+								}
+							}
+						}
+					}
+				}
+
+				choice dst_port {
+					case l4_dst_port {
+						leaf L4_DST_PORT {
+							type uint16;
+						}
+					}
+					case l4_dst_port_range {
+						leaf L4_DST_PORT_RANGE {
+							type string {
+								pattern "[0-9]{1,5}(-)[0-9]{1,5}" {
+									error-app-tag dst-port-range-invalid;
+								}
+							}
+						}
+					}
+				}
+
+				leaf TCP_FLAGS { 
+					type string {
+						pattern "0[xX][0-9a-fA-F]{2}[/]0[xX][0-9a-fA-F]{2}" {
```

Comment:
This is explicit definition.  Do you want me to put a typdef in sonic-common and use the typedef here ?

Reply:
We better define our typedef because all range/enums from standard YANG may not be supported in SONiC.

Reply:
I don't see a problem. More to follow our set guideline to use inet/standard types, wherever applicable.

Can you please explicitly explain, why we can't use the standard definition here for the flags list. This will help us to justify to others, when questioned.

Reply:
We need to take standard definition only for basic data types like int, string, ip-address, mac-address etc. 

How are we going to control if SONiC does not support a particular TCP flag defined in oc-pkt-match-types ?

Reply:
oc-pkt-match-types:TCP_FLAGS  does not match the format SONiC. oc-pkt-match-types:TCP_FLAGS is identity and SONiC looks for hex digit.

So this is comment is not applicable here.